nextflow.config

```diff
@@ -61,76 +61,191 @@ params {
     workdir = "/tmp/nextflow-work-$USER"
-    cloudDatabase = false
     cachedir = false
 }
 
 profiles {
 
-    standard { 
-        params.cloudProcess = false
-        params.cloudDatabase = false
-        includeConfig 'nextflow/configs/local.config' }
-
-    conda { 
+    //executors
+    local { 
+        workDir = params.workdir
+        params.databases = params.dbs
         params.cloudProcess = false
-        params.cloudDatabase = false
-        includeConfig 'nextflow/configs/conda.config' }
+        includeConfig 'nextflow/configs/local.config'
+    }
 
     lsf {
+        workDir = params.workdir
+        executor {
+            name = "lsf"
+            queueSize = 200
+        }        
         params.cloudProcess = true
-        includeConfig 'nextflow/configs/lsf.config' }
+        process.cache = "lenient"
+        includeConfig 'nextflow/configs/node.config'
+    }
 
     slurm {
+        workDir = params.workdir
+        executor {
+            name = "slurm"
+            queueSize = 200
+        }        
         params.cloudProcess = true
-        includeConfig 'nextflow/configs/slurm.config' }
+        process.cache = "lenient"
+        includeConfig 'nextflow/configs/node.config'
+    }
+
+    
+    //engines
+    docker { 
+        docker { enabled = true }
+        includeConfig 'nextflow/configs/container.config'
+    }
+
+    singularity {
+        singularity { 
+                enabled = true
+                autoMounts = true
+                cacheDir = params.cachedir
+        }
+        includeConfig 'nextflow/configs/container.config'
+    }
+
+    conda { 
+        includeConfig 'nextflow/configs/conda.config'
+    }
+
+
+    //pre-merged
+    standard { 
+        workDir = params.workdir
+        params.cloudProcess = false
+        includeConfig 'nextflow/configs/local.config'
+        docker { enabled = true }
+        includeConfig 'nextflow/configs/container.config'
+    }
 
     nih {
-        params.cloudProcess = true
         params.workdir = "/data/$USER/nextflow-work"
-        params.cloudDatabase = "/data/$USER/nextflow-databases/"
+        params.databases = "/data/$USER/nextflow-databases/"
         params.cachedir = "/data/$USER/nextflow-singularity/"
-        includeConfig 'nextflow/configs/slurm.config' }
+
+        workDir = params.workdir
+        executor {
+            name = "slurm"
+            queueSize = 200
+        }        
+        params.cloudProcess = true
+        process.cache = "lenient"
+        includeConfig 'nextflow/configs/node.config'
+
+        singularity { 
+                enabled = true
+                autoMounts = true
+                cacheDir = params.cachedir
+        }
+        includeConfig 'nextflow/configs/container.config' 
+    }
     
     ebi {
-        params.cloudProcess = true
   	    params.workdir = "/hps/nobackup2/production/metagenomics/$USER/nextflow-work-$USER"
```

Comment:
In the cluster it's faster to use the /scratch area for temporary files.

Reply:
ah okay! so it would be good to set the default here to 

`/scratch/nextflow-work-$USER` ? 

how is the /scratch area cleaned? Is there something like every week? 

In any case, I agree it's good to point per default to the /scratch area for the tmp files and if one really needs to resume a run in half a year it's still possible to use `--workdir` to point to another area

Reply:
The scratch are is cleaned every week.
In EBI the systems team mark that as a good practice, but being able to specify the folder is a good idea.